### PR TITLE
Support for wagtail42

### PR DESCRIPTION
**Pull request by nickmoreton**
This updates the sites dependencies so it can be used with Wagtail versions >= 4.1

I have removed the try/except and Wagtail VERSION conditionals and code that supported older versions which I hope you agree does simply the package. I understand that may not be what you prefer and I am happy to revert those changes to keep the older versions around.

I have also adjusted the tox and github actions to help simplify the running of tests.

Issues that could be solved with other updates include:
- https://github.com/jazzband/wagtailmenus/issues/435
- https://github.com/jazzband/wagtailmenus/issues/431

Some UI elements were misplaced and those are also fixed.

Also the docs could do with been updated I think but I'll wait till you let me know about supporting older versions.

### CODE REVIEW

File: wagtailmenus/conf/defaults.py [hunk 1]
```diff
@@ -89,3 +87,5 @@
 # ----------
 # Deprecated
 # ----------
+
+ADD_EDITOR_OVERRIDE_STYLES = True
```
Comment (nickmoreton) [inline wagtailmenus/conf/defaults.py:91]: These styles don't seem to be needed anymore. The wagtail UI deals with them perfectly.

File: wagtailmenus/modeladmin.py [hunk 2]
```diff
@@ -17,11 +17,6 @@ class MainMenuAdmin(ModelAdmin):
     edit_view_class = views.MainMenuEditView
     add_to_settings_menu = True
 
-    def get_form_view_extra_css(self):
```
Comment (nickmoreton) [inline wagtailmenus/modeladmin.py:20]: These styles don't seem to be needed anymore. The wagtail UI deals with them perfectly. Also removed the second instance in this file.

**Comment by djch** [comment]
Great work @nickmoreton. I hope the maintainers can merge this soon!

**Comment by schlich** [comment]
I'll take a look tonight :)

**Pull request merged**